cranelift-codegen/src/stack_layout.rs

```diff
@@ -7,15 +7,20 @@ use core::cmp::{max, min};
 
 /// Compute the stack frame layout.
 ///
-/// Determine the total size of this stack frame and assign offsets to all `Spill` and
-/// `Explicit` stack slots.
+/// Determine the total size of this stack frame and assign offsets to all `Spill` and `Explicit`
+/// stack slots.
 ///
-/// The total frame size will be a multiple of `alignment` which must be a power of two.
+/// The total frame size will be a multiple of `alignment` which must be a power of two, unless the
+/// function doesn't perform any call.
 ///
 /// Returns the total stack frame size which is also saved in `frame.frame_size`.
 ///
 /// If the stack frame is too big, returns an `ImplLimitExceeded` error.
-pub fn layout_stack(frame: &mut StackSlots, alignment: StackSize) -> CodegenResult<StackSize> {
+pub fn layout_stack(
+    frame: &mut StackSlots,
+    may_call: bool,
```

Comment:
This is a total nitpick, and maybe you disagree, but I think `is_leaf` would be a better name than `may_call` for both this parameter and the method.

Reply:
Copying my comment from Slack for posterity:

> Note that "leaf" on Windows x64 ABI has additional restrictions, though, beyond just not calling.
>
> Primarily it has to be able to unwind via a ret instruction alone.
>
> I'm good with `is_leaf`, but in the future when we have code generation tailored for Windows (primarily to exclude prologue/epilogue generation and unwind information for leafs), the nomenclature might get a little muddled.

This comment really applies to `Function.may_call` if we rename it to `Function.is_leaf`.  I think `is_leaf` here in the context of `layout_stack` makes total sense.

If we make a corresponding change to have a `Function.is_leaf`, then, as Dan suggested, perhaps it would be a good idea for a comment that makes it clear that the determination is solely based off of whether or not the function makes a call so that it isn't confused with "leaf function" in the Windows terminology.

Reply:
Good idea, renamed the function and added a comment.